sdk/modules/driver/cpp/src/camera_serialization.cpp

```diff
@@ -655,7 +655,11 @@ std::ostream &save_device(const Device &d, std::ostream &os) {
 
     google::protobuf::util::JsonPrintOptions options;
     options.add_whitespace                = true;
+    #if (GOOGLE_PROTOBUF_VERSION >= 5027000)
+    options.always_print_fields_with_no_presence = true;
```

Comment:
Do you have any documentation on this?
I don't see this field being part of the JsonPrintOptions:
https://protobuf.dev/reference/cpp/api-docs/google.protobuf.util.json_util/#JsonPrintOptions

I see the field mentionned here:
https://protobuf.dev/news/v26/#JSON
But it states that it is replacing `always_print_default_values` which is not the field we are using (`always_print_primitive_fields`)

Reply:
In regards to this change, I faced two issues while trying to build on Mac.
1. I couldn't easily come up with a package formula that allowed me to install a compatible version of Protobuf. If I can figure this out, I would remove the code changes and keep everything to < v26.
2. Once I decided to keep the newer package version, replacing the deprecated function was not straightforward. The best clues I found were from looking at the blame for PrintOptions definition (protobuf/src/google/protobuf/json/json.h) in the latest release version. In particular, these two commits ([2699579](https://github.com/protocolbuffers/protobuf/commit/26995798757fbfef5cf6648610848e389db1fecf) and [7d43131](https://github.com/protocolbuffers/protobuf/commit/7d43131a0a3a2a7640208849eeac016c6c61374e) in the [protobuf repo](https://github.com/protocolbuffers/protobuf).

TBH, I didn't love making this change, since the Protobuf documentation around it wasn't the best IMO. However, I figured it'd start a discussion as how to best resolve. I didn't spend time trying to test the implications of the new struct field, BUT I did test that the version flag worked properly.

Again would love to make this change better. Perhaps just forcing a specific version of protobuf would be best? but that also doesn't seem ideal as time goes on (right?) Just thinking out loud (please don't read anything negative in my comments!)

Reply:
Sorry for the delay on this topic, it looks like you are right. I opened a issue on protobuf side so they would update the documentation:
https://github.com/protocolbuffers/protobuf/issues/17437

So your change seems ok. However, the renaming of the field seems to have taken place in v26.0 of protobuf, so the version number to check should be `5026000`